### ISSUE TITLE
"Expected object 'MR' has no actual declaration in module <shared_debug> for JVM" (sample project included)

### ISSUE THREAD

**Issue by Mugurell** [icerockdev/moko-resources#531]
Hello,

I am trying to use `moko-resources` v`0.23.0` to be able to use "shared" strings in a compose multiplatform project but I encounter the error in the title of this issue when trying to build the application.
The project and `moko` setup I am using can be seen here - https://github.com/Mugurell/KaMPKit/commit/e628563a547bb47635595939429093adb1dcab43

Additionally if I ask `moko` to build resources I get:

```
> ./gradlew clean
> ./gradlew generateMRcommonMain --info
Task ':shared:generateMRcommonMain' is not up-to-date because:
  Output property 'outputDirectory' file /Users/petru/Projects/Tests/KaMPKit/shared/build/generated/moko/commonMain/src/co has been removed.
  Output property 'outputDirectory' file /Users/petru/Projects/Tests/KaMPKit/shared/build/generated/moko/commonMain/src/co/touchlab has been removed.
  Output property 'outputDirectory' file /Users/petru/Projects/Tests/KaMPKit/shared/build/generated/moko/commonMain/src/co/touchlab/kampkit has been removed.
file or directory '/Users/petru/Projects/Tests/KaMPKit/shared/build/generated/moko/commonMain/res', not found
file or directory '/Users/petru/Projects/Tests/KaMPKit/shared/build/generated/moko/commonMain/res', not found
file or directory '/Users/petru/Projects/Tests/KaMPKit/shared/build/generated/moko/commonMain/res', not found
file or directory '/Users/petru/Projects/Tests/KaMPKit/shared/build/generated/moko/commonMain/res', not found
file or directory '/Users/petru/Projects/Tests/KaMPKit/shared/build/generated/moko/commonMain/res', not found
file or directory '/Users/petru/Projects/Tests/KaMPKit/shared/build/generated/moko/commonMain/res', not found
file or directory '/Users/petru/Projects/Tests/KaMPKit/shared/build/generated/moko/commonMain/res', not found

```

with the `expect` class constructed
but not also with the `actual` ones
![image](https://github.com/icerockdev/moko-resources/assets/11428869/9e3022b9-ad29-4479-8714-210c39b8762f)

**Comment by izadiegizabal** [comment]
i just encountered this issue when upgrading my project's AGP to `8.2.0-alpha13` and kotlin to `1.9.0`, it was working before. i wonder if the kotlin update broke it 🤔

**Comment by alexzhirkevich** [comment]
Had the same issue recently after doing some stuff with gradle. Event git rollback to the point where everything was ok didn't help =)))).

The only solution i found is to clear project and completely delete `$USER_HOME/.gradle` folder.

**Comment by izadiegizabal** [comment]
i just tried cleaning the project, invalidating caches and purging `$USER_HOME/.gradle` but that didn't work, unfortunately.
in my case rolling kotlin back to `1.8.22` seems to prevent the error though 🤔

**Comment by ExNDY** [comment]
I think this's continuation of problem (or its variant): https://github.com/icerockdev/moko-resources/issues/353#issuecomment-1179713713
Worked solution: https://github.com/icerockdev/moko-resources/issues/510#issuecomment-1619141070
Tested on AGP: 8.0.1, Gradle: 8.2.1, Kotlin 1.9.0

**Comment by Mugurell** [comment]
Thank you!
Was trying various mix and matches of kotlin, agp and gradle but seems to indeed be just a linking issue that hopefully will get fixed in the next versions.

![image](https://github.com/icerockdev/moko-resources/assets/11428869/3a3e1874-ef65-4f70-bcb3-c127c7fdb7f7)

**Comment by Alex009** [comment]
issue will be fixed so we should not close this issue now :)

**Comment by mikolajefento** [comment]
For me the fix was to add `dependsOn(commonMain)` in `androidMain` source set

**Comment by SongOnWater** [comment]
> i just encountered this issue when upgrading my project's AGP to `8.2.0-alpha13` and kotlin to `1.9.0`, it was working before. i was wondering if the kotlin update broke it but i see that you are using `1.8.22` 🤔

This issue is due to kotlin plugin 1.9.0. But maybe another plugin(such as com.arkivanov.parcelize.darwin 0.2.0) has used the 1.9.0 version, the higher version has the priority. Y

**Comment by m-sasha** [comment]
Is this fixed on the moko-resources side?

I'm with the Compose-Multiplatform team, trying to understand whether this is the same issue [reported here](https://github.com/JetBrains/compose-multiplatform/issues/3603) and whether any action on our part is required.

**Comment by bcmedeiros** [comment]
In here, after adding a `dependsOn(commonMain)` to `androidMain` source set, my errors disappeared.

**Comment by hellomr3** [comment]
> my

it work for me

**Comment by maitriyogin** [comment]
I'm getting this issue with a fresh CMP template with kotlin 1.9.21.
dependsOn is no longer with the new Gradle setup so instead I've tried 
```
 androidMain.dependencies {
            implementation(project(":shared"))
```

**Comment by lindsey-hughes** [comment]
> I'm getting this issue with a fresh CMP template with kotlin 1.9.21. dependsOn is no longer with the new Gradle setup so instead I've tried
> 
> ```
>  androidMain.dependencies {
>             implementation(project(":shared"))
> ```
> 
> Although this doesn't seem to work ... Any ideas?

@maitriyogin adding this line worked for me
```
androidMain {
    dependsOn(commonMain.get())
}
```

**Comment by tahaak67** [comment]
> > I'm getting this issue with a fresh CMP template with kotlin 1.9.21. dependsOn is no longer with the new Gradle setup so instead I've tried
> > ```
> >  androidMain.dependencies {
> >             implementation(project(":shared"))
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Although this doesn't seem to work ... Any ideas?
> 
> @maitriyogin adding this line worked for me
> 
> ```
> androidMain {
>     dependsOn(commonMain.get())
> }
> ```

Thanks after adding `dependsOn(commonMain.get())` android build is running fine but it introduces errors with ios. if i comment the 'dependsOn' line and sync ios build works fine but the issue in op comes back on android.

**Comment by titanseason** [comment]
> > @maitriyogin adding this line worked for me
> > ```
> > androidMain {
> >     dependsOn(commonMain.get())
> > }
> > ```
> 
> Thanks after adding `dependsOn(commonMain.get())` android build is running fine but it introduces errors with ios. if i comment the 'dependsOn' line and sync ios build works fine but the issue in op comes back on android.

I got the same issue  _after adding `dependsOn(commonMain.get())`_, iosMain not working

**Comment by 3llomi** [comment]
> > > @maitriyogin adding this line worked for me
> > > ```
> > > androidMain {
> > >     dependsOn(commonMain.get())
> > > }
> > > ```
> > 
> > 
> > Thanks after adding `dependsOn(commonMain.get())` android build is running fine but it introduces errors with ios. if i comment the 'dependsOn' line and sync ios build works fine but the issue in op comes back on android.
> 
> I got the same issue _after adding `dependsOn(commonMain.get())`_, iosMain not working

Same here :(

**Comment by magdalenatsolaki** [comment]
kotlin = "1.9.23"
moko = "0.23.0"
compose = "1.6.1"
....

worked for me:

```
androidMain {
    dependsOn(commonMain)
}
```

**Comment by ExNDY** [comment]
will be fixed in 0.24.0, please check in 0.24.0-alpha-6
